### PR TITLE
(jkube-kit/jkube-kit-wildfly-jar/) : Migrate tests from JMockit to Mockito #1729

### PR DESCRIPTION
**Pull request by baruKreddy**
Signed-off-by: baruKreddy [bhargavirkotha@gmail.com](mailto:bhargavirkotha@gmail.com)

Fixes https://github.com/eclipse/jkube/issues/1602

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [ ] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [ ] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [ ] My code follows the style guidelines of this project
 - [ ] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [ ] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [ ] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by baruKreddy** [comment]
@manusa @rohanKanojia Have created a new PR since old PR has been reverting Anurag's changes. Please review :)

File: jkube-kit/jkube-kit-wildfly-jar/src/test/java/org/eclipse/jkube/wildfly/jar/generator/WildflyJARGeneratorTest.java [hunk 1]
```diff
@@ -27,154 +27,163 @@
...
+        Map<String, Object> options = new HashMap<>();
+        Map<String, String> pluginOptions = new HashMap<>();
+        options.put(PLUGIN_OPTIONS, pluginOptions);
+        pluginOptions.put(JBOSS_MAVEN_DIST, null);
+        pluginOptions.put(JBOSS_MAVEN_REPO, "target" + File.separator + "myrepo");
+        //
+        Path tmpDir = Files.createTempDirectory("bootable-jar-test-project");
+        Path targetDir = tmpDir.resolve("target");
+        Path repoDir = targetDir.resolve("myrepo");
+        Files.createDirectories(repoDir);
```
Comment (manusa) [inline jkube-kit/jkube-kit-wildfly-jar/src/test/java/org/eclipse/jkube/wildfly/jar/generator/WildflyJARGeneratorTest.java:88]: This was OK as it was, also, the finally clause deleting these files is not necessary. The extensions takes care of this.

File: jkube-kit/jkube-kit-wildfly-jar/src/test/java/org/eclipse/jkube/wildfly/jar/generator/WildflyJARGeneratorTest.java [hunk 2]
```diff
@@ -27,154 +27,163 @@
...
+import static org.mockito.Mockito.when;
 
 /**
  * @author roland
  */
 
 class WildflyJARGeneratorTest {
 
-    @TempDir
-    Path temporaryFolder;
```
Comment (manusa) [inline jkube-kit/jkube-kit-wildfly-jar/src/test/java/org/eclipse/jkube/wildfly/jar/generator/WildflyJARGeneratorTest.java:49]: Please keep using this extension which avoids the usage of the redundant finally clause you added into some test cases

Comment (baruKreddy) [reply jkube-kit/jkube-kit-wildfly-jar/src/test/java/org/eclipse/jkube/wildfly/jar/generator/WildflyJARGeneratorTest.java]: Made requested changes :)

File: jkube-kit/jkube-kit-wildfly-jar/src/test/java/org/eclipse/jkube/wildfly/jar/generator/WildflyJARGeneratorTest.java [hunk 3]
```diff
@@ -228,40 +227,31 @@ void slimServerTrueDist(@Mocked final JavaProject project) {
...
-            context.getProject(); result = project;
-            project.getOutputDirectory(); result = temporaryFolder.toFile();
-            project.getPlugins(); result = Collections.emptyList(); minTimes = 0;
-            project.getVersion(); result = "1.0.0"; minTimes = 0;
-        }};
+
+    private GeneratorContext createGeneratorContext() throws IOException {
+        when(context.getProject()).thenReturn(project);
+        String tempDir = Files.createTempDirectory("wildfly-jar-test-project").toFile().getAbsolutePath();
+        when(project.getOutputDirectory()).thenReturn(new File(tempDir));
```
Comment (rohanKanojia) [inline jkube-kit/jkube-kit-wildfly-jar/src/test/java/org/eclipse/jkube/wildfly/jar/generator/WildflyJARGeneratorTest.java:252]: Why was this change required? Was `temporaryFolder.toFile()` causing problems with mockito?

File: jkube-kit/jkube-kit-wildfly-jar/src/test/java/org/eclipse/jkube/wildfly/jar/generator/WildflyJARGeneratorTest.java [hunk 4]
```diff
@@ -48,133 +48,132 @@ class WildflyJARGeneratorTest {
...
-              .isEqualTo("-Djava.net.preferIPv4Stack=true"))
-          .satisfies(option -> assertThat(option).last()
-              .isEqualTo("-Dmaven.repo.local=/deployments/myrepo"));
+    void slimServerAbsoluteDir() throws IOException {
+        Map<String, Object> options = new HashMap<>();
+        Map<String, String> pluginOptions = new HashMap<>();
+        Path tmpDir = Files.createTempDirectory("bootable-jar-test-project2");
+        Path targetDir = tmpDir.resolve("target");
+        Path repoDir = targetDir.resolve("myrepo");
+        Files.createDirectories(repoDir);
```
Comment (rohanKanojia) [inline jkube-kit/jkube-kit-wildfly-jar/src/test/java/org/eclipse/jkube/wildfly/jar/generator/WildflyJARGeneratorTest.java:120]: I'm not able to understand the purpose behind these path-related changes. I think if we just stick to mockito statements, test should work.

**Comment by baruKreddy** [comment]
Looks good to be reviewed!

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]
